python/ppml/example/fgboost_regression/fgboost-tutorial.md

```diff
@@ -0,0 +1,172 @@
+# Tutorial for Vertically Federated XGBoost
+This example provides a step-by-step tutorial of running a FGBoost Regression task with 2 parties.
+## 1. Key Concepts
+### 1.1 FGBoost 
+**FGBoost** stands for Federated Gradient Boosted Tree algorithm. It allows multiple parties to run a federated gradient boosted decision tree application.
+### 1.2 PSI
+**PSI** stands for Private Set Intersection algorithm. It compute the intersection of data from different parties and return the result so that parties know which data they could collaborate with.
+### 1.3 SGX
+**SGX** stands for [Software Guard Extensions](https://www.intel.com/content/www/us/en/developer/tools/software-guard-extensions/overview.html). It helps protect data in use via unique application isolation technology. The data of application running in SGX would be protected.
+
+### 1.4 FL Server and Client
+A **FL Server** is a gRPC server to handle requests from FL Client. A **FL Client** is a gRPC client to send requests to FL Server. These requests include:
+* serialized model to use in training at FL Server
+* some model related instance, e.g. loss function, optimizer
+* the Tensor which FL Server and FL Client interact, e.g. predict output, label, gradient
+
+
+## 2. Build FGBoost Client Applications
+This section introduces the details of the example code.
+
+We use [House Prices](https://www.kaggle.com/competitions/house-prices-advanced-regression-techniques/data) dataset. To simulate the scenario where different data features are held by 2 parties respectively, we split the dataset to 2 parts. The split is taken by select every other column (code at [split script](scala/ppml/scripts/split_dataset.py)).
+
+The code is available in projects, including [Client 1 code](fgboost_regression_party_1.py) and [Client 2 code](fgboost_regression_party_2.py). You could directly start two different terminals are run them respectively to start a federated learning, and the order of start does not matter. Following is the detailed step-by-step tutorial to introduce how the code works.
+
+### 2.1 FL Context
+First, import the package and initilize FL Context.
+```python
+from bigdl.ppml.fl import *
+from bigdl.ppml.fl.algorithms.fgboost_regression import FGBoostRegression
+import pandas as pd
+
+init_fl_context()
```

Comment:
is there a client id?

Reply:
Oh this is missed.. Client ID is only regulated in NN FLServer.. These two are actually two different implementation.. will add soon

Reply:
@jason-dai just figured out that FGBoost does not require order info. So do we still need to regulate Client ID?

Reply:
We should probably keep the interface consistent?